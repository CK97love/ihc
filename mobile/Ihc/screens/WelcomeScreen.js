import React, { Component } from 'react';
import {
  Platform,
  StyleSheet,
  Button,
  Text,
  View
} from 'react-native';

export default class WelcomeScreen extends Component<{}> {
  constructor(props) {
    super(props);
  }

  goToSignin = () => {
    this.props.navigator.push({
      screen: 'Ihc.SigninScreen',
      title: 'Signin'
    });
  }

  goToSelectPatient = () => {
    this.props.navigator.push({
      screen: 'Ihc.PatientSelectScreen',
      title: 'Select patient'
    });
  }

  // TODO: Download updates from server and add them to local storage
  sync = () => {
    // TODO: Delete, temp for testing:
    this.props.navigator.push({
      screen: 'Ihc.GrowthChartScreen',
      title: 'Growth Chart'
    });
  }

  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.welcome}>
          Welcome to clinic!
        </Text>
        <Button onPress={this.goToSignin}
          title="Signin"
        />
        <Button onPress={this.goToSelectPatient}
          title="Select Patient"
        />
        <Button onPress={this.sync}
          title="Sync"
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  instructions: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5,
  },
});
