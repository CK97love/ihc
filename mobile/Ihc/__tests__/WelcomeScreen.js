import 'react-native';
import React from 'react';
import WelcomeScreen from '../screens/WelcomeScreen';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const json = renderer.create(
    <WelcomeScreen />
  ).toJSON();
  expect(json).toMatchSnapshot();
});
