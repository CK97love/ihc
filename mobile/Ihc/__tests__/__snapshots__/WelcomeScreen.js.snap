// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F5FCFF",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "flex": 0,
        "justifyContent": "center",
        "padding": 20,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "fontSize": 20,
            "margin": 10,
            "textAlign": "center",
          }
        }
      >
        Welcome to clinic!
      </Text>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2196F3",
            "borderRadius": 2,
            "elevation": 4,
            "height": 40,
            "justifyContent": "center",
            "margin": 4,
            "opacity": 1,
            "padding": 8,
            "width": 140,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fefefe",
                "fontWeight": "500",
                "textAlign": "center",
              },
              undefined,
            ]
          }
        >
          Signin
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2196F3",
            "borderRadius": 2,
            "elevation": 4,
            "height": 40,
            "justifyContent": "center",
            "margin": 4,
            "opacity": 1,
            "padding": 8,
            "width": 140,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fefefe",
                "fontWeight": "500",
                "textAlign": "center",
              },
              undefined,
            ]
          }
        >
          Select Patient
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2196F3",
            "borderRadius": 2,
            "elevation": 4,
            "height": 40,
            "justifyContent": "center",
            "margin": 4,
            "opacity": 1,
            "padding": 8,
            "width": 140,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fefefe",
                "fontWeight": "500",
                "textAlign": "center",
              },
              undefined,
            ]
          }
        >
          Upload updates
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2196F3",
            "borderRadius": 2,
            "elevation": 4,
            "height": 40,
            "justifyContent": "center",
            "margin": 4,
            "opacity": 1,
            "padding": 8,
            "width": 140,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fefefe",
                "fontWeight": "500",
                "textAlign": "center",
              },
              undefined,
            ]
          }
        >
          Download updates
        </Text>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F5FCFF",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "red",
          "textAlign": "center",
        }
      }
    />
    <Text
      style={
        Object {
          "color": "green",
          "textAlign": "center",
        }
      }
    />
  </View>
</View>
`;
